### ISSUE TITLE
After #959 operator does not update sidecar arguments

### ISSUE THREAD

**Issue by ctron** [jaegertracing/jaeger-operator#1053]
After upgrading, and having the fix for #959, the TLS configuration was still broken, as the controller did not update the arguments of the sidecar.

Deleting the deployment triggered the re-creation of the side and brought back tracing.

**Comment by objectiser** [comment]
@ctron Hi. I assume you are running on OpenShift?

Can you confirm the version you were upgrading from?

**Comment by ctron** [comment]
> @ctron Hi. I assume you are running on OpenShift?

Yes, indeed :grin: 

> Can you confirm the version you were upgrading from?

I can't really tell. The status section still reports it is `1.1.16.0`, which is another bug I guess. Now it is `1.17.1`. But I am not sure what was in the middle.

**Comment by objectiser** [comment]
Ok thanks - we will try to reproduce.

**Comment by jpkrohling** [comment]
cc @rubenvp8510

**Comment by tux-o-matic** [comment]
I experience something similar while running the Jaeger Operator v 1.17.3 in OCP 4.3. If a Pod with the Jaeger sidecar also happens to have the Istio sidecar.
Even with gRPC configured to use TLS on the collector and injected sidecar, as mentioned in #959, I see:
```shell
transport: authentication handshake failed: tls: first record does not look like a TLS handshake
```
But if I remove *spec.template.metadata.annotations sidecar.istio.io/inject: 'true'* from my Deployment then I no longer see errors in between the Jaeger agent sidecar and the collector.

**Comment by objectiser** [comment]
@tux-o-matic I think this sounds like a separate issue - nothing to do with sidecar injected agent flags not being updated on upgrade. Could you create a separate issue?
